### PR TITLE
Added refresh option for subordinate charms.

### LINKED ISSUE #61: Subordinates are not able to make ch migration
Currently, the logic implemented on `OpenStackSubordinateApplication` with `_default_used` and the channel setter doesn't allow the application to make the migration if necessary.

### PR DESCRIPTION
**Pull request by esunar**
resolves #61

### CODE REVIEW

**Review by rgildein — APPROVED** [review]
Code looks good to me, but I  do not understand how this solved the issue or what is actual issue. So more reviews are needed.

**Review by gabrielcocenza — CHANGES_REQUESTED** [review]
It seems that this solves the issue, but please add a unit test simulating the ch migration for subordinates apps.

**Comment by esunar** [comment]
> It seems that this solves the issue, but please add a unit test simulating the ch migration for subordinates apps.

done

File: tests/unit/apps/test_subordinate.py [hunk 1]
```diff
@@ -96,3 +96,56 @@ def test_channel_setter_invalid(status, channel):
 
     app.channel = channel
     assert app.channel == "ussuri/stable"
+
+
+@pytest.mark.parametrize(
+    "channel",
+    [
+        "focal/edge",
```
Comment (gabrielcocenza) [inline tests/unit/apps/test_subordinate.py:104]: I think those channels are a little bit misleading. IIRC, when charms comes from cs it always will be set to `stable`. E.g:

```yaml
applications:
  ubuntu:
    charm: cs:ubuntu-24
    series: bionic
    os: ubuntu
    charm-origin: charmstore
    charm-name: ubuntu
    charm-rev: 24
    charm-channel: stable
    exposed: false
```

So there is no need to have a parameterize function with different channels.

File: tests/unit/apps/test_subordinate.py [hunk 2]
```diff
@@ -96,3 +96,56 @@ def test_channel_setter_invalid(status, channel):
...
+        "latest/edge",
+        "latest/stable",
+        "something/stable",
+    ],
+)
+def test_generate_plan(status, channel):
```
Comment (gabrielcocenza) [inline tests/unit/apps/test_subordinate.py:110]: `test_generate_plan_ch_migration`

File: tests/unit/apps/test_subordinate.py [hunk 3]
```diff
@@ -96,3 +96,56 @@ def test_channel_setter_invalid(status, channel):
...
+    app.channel = channel
+    plan = app.generate_upgrade_plan("wallaby")
+    assert str(plan) == (
+        "Upgrade plan for 'my_keystone_ldap' to wallaby\n"
+        "\tMigration of 'my_keystone_ldap' from charmstore to charmhub\n"
+        "\tUpgrade 'my_keystone_ldap' to the new channel: 'wallaby/stable'\n"
+    )
+
+
+def test_generate_plan_stable1(status):
```
Comment (gabrielcocenza) [inline tests/unit/apps/test_subordinate.py:125]: I think we can remove this test once the suggestions above are done

File: tests/unit/apps/test_subordinate.py [hunk 4]
```diff
@@ -96,3 +96,56 @@ def test_channel_setter_invalid(status, channel):
...
+    app.channel = "ussuri/stable"
+    plan = app.generate_upgrade_plan("wallaby")
+    assert str(plan) == (
+        "Upgrade plan for 'my_keystone_ldap' to wallaby\n"
+        "\tMigration of 'my_keystone_ldap' from charmstore to charmhub\n"
+        "\tUpgrade 'my_keystone_ldap' to the new channel: 'wallaby/stable'\n"
+    )
+
+
+def test_generate_plan_stable2(status):
```
Comment (gabrielcocenza) [inline tests/unit/apps/test_subordinate.py:140]: rename this test to something like `test_generate_plan_victoria_to_wallaby`

File: tests/unit/apps/test_subordinate.py (hunk 1 above)
Comment (gabrielcocenza) [reply tests/unit/apps/test_subordinate.py:104]: I saw that conftest is alreasy setting the channel right as I said here

**Review by gabrielcocenza — CHANGES_REQUESTED** [review]

File: tests/unit/apps/test_subordinate.py (hunk 1 above)
Comment (esunar) [reply tests/unit/apps/test_subordinate.py]: made some improvements

File: tests/unit/apps/test_subordinate.py (hunk 2 above)
Comment (esunar) [reply tests/unit/apps/test_subordinate.py]: done

File: tests/unit/apps/test_subordinate.py (hunk 3 above)
Comment (esunar) [reply tests/unit/apps/test_subordinate.py]: removed

File: tests/unit/apps/test_subordinate.py (hunk 4 above)
Comment (esunar) [reply tests/unit/apps/test_subordinate.py]: added from_to list for plan generation

**Review by gabrielcocenza — APPROVED** [review]
LGTM. Non-blocker:I think with the new tests added we can remove [test_generate_upgrade_plan](https://github.com/canonical/charmed-openstack-upgrader/blob/main/tests/unit/apps/test_subordinate.py#L43)

**Review by aieri — APPROVED** [review]
Looks good, but please ensure the commit message explains why this is necessary

**Pull request merged**